### ISSUE TITLE
Connect to the docker API from the management network

### ISSUE THREAD

**Issue by hmahmood** [vmware/vic#6339]
@JohnMurrayUK commented on [Thu Sep 14 2017](https://github.com/vmware/vic-product/issues/819)

**VIC Product version:**

OVA version vic-b6d8345b (post 1.2.0 rc3)

**Deployment details:**

Typical secure multi tenant service provider environment.  No access from client vlans to vSphere management vlans via routing or firewalls.  All client interaction with vCenter is performed out of band via a service portal.  One ova is deployed to support all tenants, but each tenant has multiple VCH appliances available to it. 

**Steps to reproduce:**

Deploy the OVA to the management network.  Same subnet as vCenter and ESXi hosts.
Deploy a VCH to a tenant with an IP address on the client's vlan and a management interface on the management network using --management-network-ip switch so it can deploy containers by being dual homed.

**Actual behavior:**

Unable to add the VCH to the ova portal as ports :2375-2378 are not listening on the management interface.

**Expected behavior:**

The listeners on tcp2375-2378 should be listening on ALL interfaces so the service provider can add the vch to the ova portal via the management interface to allow them to assign permissions, and define tenants as projects (would be better to define tenants explicitly instead of calling them projects, but thats a different issue).

**Additional details as necessary:**

We cannot allow client networks to 'route' to the management network as many tenants use the same subnets on their NSX vxlans.
We do not allow direct access from tenant networks to the management network.
We can work with VCH's being dual homed as they can only be deployed by the service provider team, and we can restrict access via NSX firewall and vCenter roles of vch opsuser to prevent backdoor access to the management network.
VCH in our case does not need a route/gateway for the management interface, as it is a flat subnet, but that may not be the case for all service providers, so a --management-network-static-route command might be useful.

**Comment by anchal-agrawal** [comment]
@hmahmood Please add an estimate, priority and team labels as needed, thanks!

**Comment by hickeng** [comment]
I'm not happy with exposing docker API or vicadmin on the public interface - that's explicitly intended for application access via port forwarding if not using container-network.

@johnmurrayuk I'm curious as to why the OVA is deployed on the management network - is this purely so it has access to the PSC? The issue description says `No access from client vlans to vSphere management vlans via routing or firewalls` - how are clients connecting to the OVA?
As a forward looking comment, we're working towards the original VCH architecture plan that doesn't have a NIC connected to the management network at all in support of the management/VM separation best practice.

**Comment by JohnMurrayUK** [comment]
Our tenants have autonomy over their addressing schemes, and their security.  As such we are unable to route or nat their networks to our management vlan.  All their interaction with our vCenter is out-of-band via a separate web portal.  As we understood there could only be one OVA appliance, per vCenter (now clarified as not a limitation), and the portal could not be dual homed, we had to deploy the ova portal on the management network. Thus when we deploy a VCH to one of our tenants, with the additional management NIC, all worked fine apart from we could not connect the vch to the ova as the ova was searching for a port which was not listening on the vch's management nic.
There are two possible solutions to this that I can see. 1. allow the vch to listen for connections from the ova via its management interface.  2. allow the ova to be dual homed so we can deploy it to the tenant network (now that multiple ova's are supported).

**Comment by mhagen-vmware** [comment]
@hickeng Please resolve all dependencies and move this into the backlog or re-assign it to the person that needs to get this story out of not ready.

**Comment by hickeng** [comment]
If implementing this via a change to the VCH I _suggest_ that we do this by providing a list of other network roles that the API can listen on. We can do the same thing for the portlayer as well which would provide benefit in local testing and debugging.

I strongly prefer not to dual home the OVA in its current form - it would provide a management/vm network bridge and is a much larger attack surface to secure than the VCH code.

I _require_ that we do _not_ have the VCH API servers listening on the public interface (when separate from other roles) unless directly explicit as part of the install. 

The network roles all have names in the endpointVM /etc/hosts file which should be used for socket binding:
```
client.localhost
management.localhost
bridge.localhost
public.localhost
```

**Comment by sflxn** [comment]
@hickeng This sounds like there is work in both API servers and also vic-machine/ui wizard.

**Comment by gigawhitlocks** [comment]
TL;DR:
>We need to get the persona and portlayer to listen to one of the other network roles (with restriction)

>we also need the UI wizard to allow picking which one

**Comment by mdubya66** [comment]
This needs more investigation and thought. Removing from 1.3 and changing to an enhancement.

**Comment by mdubya66** [comment]
approved by @pdaigle

**Comment by zjs** [comment]
Closing this in favor of vmware/vic-product#940.
